src/Topic/TopicManager.php

```diff
@@ -76,11 +76,13 @@ public function onPublish(ConnectionInterface $conn, $topic, $event, array $excl
 
     public function onClose(ConnectionInterface $conn): void
     {
-        $this->app->onClose($conn);
-
         foreach ($this->topicLookup as $topic) {
             $this->cleanTopic($topic, $conn);
+
+            $this->app->onUnsubscribe($conn, $topic);
```

Comment:
I really wish I kept better notes from 2 years ago, but #422 added this same line and I ended up having to revert it after that PR made it to production for one of my clients.  I don't remember what happened exactly (and our Sentry logs are long purged from that rollout), or if that ended up being an incomplete fix for what you're working on here, but figured I'd point it out since it came to mind.

Reply:
Ah, yes I think it's the same issue, let me take a look at what happened there.

Reply:
@mbabker I think the issue was that he didn't remove this line, so `onUnsubscribe` got called twice:
https://github.com/GeniusesOfSymfony/WebSocketBundle/blob/3a8b6a0a78234f2f38fe2168ec2df423aa7d3950/src/Server/App/WampApplication.php#L177

Reply:
Yeah, that would make sense.

Reply:
After his change, the order was: `onUnsubscribe`, `onClose`, `cleanTopic`, `onUnsubscribe`

Reply:
Since `cleanTopic` clears `$conn->WAMP->subscriptions`, it was necessary to remove the call to `$this->topicDispatcher->onUnSubscribe` and call it through `$this->app->onUnsubscribe($conn, $topic);` instead.